### PR TITLE
etcd-tester: retry tester clients calls

### PR DESCRIPTION
**Pull request by gyuho**
For https://github.com/coreos/etcd/issues/4517.

### CODE REVIEW

File: tools/functional-tester/etcd-tester/cluster.go [hunk 1]
```diff
@@ -266,12 +266,23 @@ func setHealthKey(us []string) error {
 		if err != nil {
 			return fmt.Errorf("%v (%s)", err, u)
 		}
-		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
 		kvc := pb.NewKVClient(conn)
-		_, err = kvc.Put(ctx, &pb.PutRequest{Key: []byte("health"), Value: []byte("good")})
-		cancel()
+		for i := 0; i < 5; i++ {
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/cluster.go]: there are a lot of code duplication that can be reduced easily. can you make the retry a func?

File: tools/functional-tester/etcd-tester/cluster.go [hunk 2]
```diff
@@ -266,12 +266,23 @@ func setHealthKey(us []string) error {
...
+			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+			_, err = kvc.Put(ctx, &pb.PutRequest{Key: []byte("health"), Value: []byte("good")})
+			cancel()
+			if grpc.ErrorDesc(err) == context.DeadlineExceeded.Error() {
+				log.Printf("etcd-tester: kvc.Put retrying with %v", err)
+				time.Sleep(time.Second)
+				continue
+			}
+			if err == nil {
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/cluster.go]: check the error == nil case before the deadline case. usually we want to handle simple and most common case first.

File: tools/functional-tester/etcd-tester/cluster.go (hunk 1 above)
Comment (gyuho) [inline tools/functional-tester/etcd-tester/cluster.go]: Sure, I will make this simpler. Thanks.

File: tools/functional-tester/etcd-tester/cluster.go (hunk 2 above)
Comment (gyuho) [inline tools/functional-tester/etcd-tester/cluster.go]: Got it. Thanks.

**Comment by gyuho** [comment]
@xiang90 Wrapped common operations with `retry`. PTAL. Thanks.

**Comment by xiang90** [comment]
i do not feel this is necessary. It seems that we already found the root cause of the errors and do not need to retry anymore. Shall we close this?

**Comment by gyuho** [comment]
@xiang90 Agree. Thanks.

**Pull request closed without merging**